src/Instrumentation/AwsSdk/AwsSdkInstrumentation.php

```diff
@@ -37,8 +55,85 @@ public function init(): bool
         return true;
     }
 
+    public function setPropagator(TextMapPropagatorInterface $propagator): void
+    {
+        $this->propagator = $propagator;
+    }
+
+    public function getPropagator(): TextMapPropagatorInterface
+    {
+        return $this->propagator;
+    }
+
+    public function setTracerProvider(TracerProviderInterface $tracerProvider): void
+    {
+        $this->tracerProvider = $tracerProvider;
+    }
+
+    public function getTracerProvider(): TracerProviderInterface
+    {
+        return $this->tracerProvider;
+    }
+
+    public function getTracer(): TracerInterface
+    {
+        return $this->tracerProvider->getTracer('io.opentelemetry.contrib.php');
+    }
+
+    public function getContext():Context
+    {
+        return Context::getRoot();
+    }
+
+    public function instrumentClients($clientsArray) : void
+    {
+        $this->clients = $clientsArray;
+    }
+
     public function activate(): bool
     {
+        try {
+            $middleware = Middleware::tap(function ($cmd, $req) {
+                $tracer = $this->getTracer();
+                $propagator = $this->getPropagator();
+
+                $carrier = [];
+
+                $this->span = $tracer->spanBuilder($this->clientName . '.' . $cmd->getName())->setSpanKind(AwsSdkInstrumentation::SPAN_KIND)->startSpan();
+                $this->scope = $this->span->activate();
+
+                $propagator->inject($carrier, null, $this->getContext());
```

Comment:
I'm not totally sure that the root context is what you want here. We generally use current context (which the propagator will use by default)